server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/webhook/WebhookPayload.java

```diff
@@ -0,0 +1,103 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.task.projectanalysis.webhook;
+
+import java.io.StringWriter;
+import java.io.Writer;
+import java.util.Date;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import org.sonar.api.ce.posttask.CeTask;
+import org.sonar.api.ce.posttask.PostProjectAnalysisTask;
+import org.sonar.api.ce.posttask.Project;
+import org.sonar.api.ce.posttask.QualityGate;
+import org.sonar.api.utils.text.JsonWriter;
+
+import static java.util.Objects.requireNonNull;
+
+public class WebhookPayload {
+
+  private final String projectKey;
+  private final String json;
+
+  public WebhookPayload(String projectKey, String json) {
+    this.projectKey = requireNonNull(projectKey);
+    this.json = requireNonNull(json);
+  }
+
+  public String getProjectKey() {
+    return projectKey;
+  }
+
+  public String toJson() {
+    return json;
+  }
+
+  public static WebhookPayload from(PostProjectAnalysisTask.ProjectAnalysis analysis) {
+    Writer string = new StringWriter();
+    JsonWriter writer = JsonWriter.of(string);
+    writer.beginObject();
+    writeTask(writer, analysis.getCeTask());
+    Optional<Date> analysisDate = analysis.getAnalysisDate();
+    if (analysisDate.isPresent()) {
+      writer.propDateTime("analysedAt", analysisDate.get());
+    }
+    writeProject(analysis, writer, analysis.getProject());
+    writeQualityGate(writer, analysis.getQualityGate());
+    writer.endObject().close();
+    return new WebhookPayload(analysis.getProject().getKey(), string.toString());
+  }
+
+  private static void writeTask(JsonWriter writer, CeTask ceTask) {
+    writer.prop("taskId", ceTask.getId());
+    writer.prop("status", ceTask.getStatus().toString());
+  }
+
+  private static void writeProject(PostProjectAnalysisTask.ProjectAnalysis analysis, JsonWriter writer, Project project) {
+    writer.name("project");
+    writer.beginObject();
+    writer.prop("key", project.getKey());
+    writer.prop("name", analysis.getProject().getName());
+    writer.endObject();
+  }
+
+  private static void writeQualityGate(JsonWriter writer, @Nullable QualityGate gate) {
+    if (gate != null) {
+      writer.name("qualityGate");
+      writer.beginObject();
+      writer.prop("name", gate.getName());
+      writer.prop("status", gate.getStatus().toString());
+      writer.name("conditions").beginArray();
+      for (QualityGate.Condition condition : gate.getConditions()) {
+        writer.beginObject();
+        writer.prop("metric", condition.getMetricKey());
+        writer.prop("operator", condition.getOperator().name());
+        writer.prop("value", condition.getValue());
+        writer.prop("status", condition.getStatus().name());
```

Comment:
Tested on a blank SQ instance with a global webhook to a non existing URL.

The task failed on this line with:
```java
java.lang.IllegalStateException: There is no value when status is NO_VALUE
	at com.google.common.base.Preconditions.checkState(Preconditions.java:173)
	at org.sonar.server.computation.task.projectanalysis.api.posttask.ConditionImpl.getValue(ConditionImpl.java:166)
	at org.sonar.server.computation.task.projectanalysis.webhook.WebhookPayload.writeQualityGate(WebhookPayload.java:92)
	at org.sonar.server.computation.task.projectanalysis.webhook.WebhookPayload.from(WebhookPayload.java:63)
	at org.sonar.server.computation.task.projectanalysis.webhook.WebhookPostTask.process(WebhookPostTask.java:60)
	at org.sonar.server.computation.task.projectanalysis.webhook.WebhookPostTask.finished(WebhookPostTask.java:50)
	at org.sonar.server.computation.task.projectanalysis.api.posttask.PostProjectAnalysisTasksExecutor.executeTask(PostProjectAnalysisTasksExecutor.java:106)
	at org.sonar.server.computation.task.projectanalysis.api.posttask.PostProjectAnalysisTasksExecutor.finished(PostProjectAnalysisTasksExecutor.java:100)
	at org.sonar.server.computation.task.step.ComputationStepExecutor.executeListener(ComputationStepExecutor.java:71)
	at org.sonar.server.computation.task.step.ComputationStepExecutor.execute(ComputationStepExecutor.java:56)
	at org.sonar.server.computation.task.projectanalysis.taskprocessor.ReportTaskProcessor.process(ReportTaskProcessor.java:75)
	at org.sonar.server.computation.taskprocessor.CeWorkerCallableImpl.executeTask(CeWorkerCallableImpl.java:84)
	at org.sonar.server.computation.taskprocessor.CeWorkerCallableImpl.call(CeWorkerCallableImpl.java:57)
	at org.sonar.server.computation.taskprocessor.CeWorkerCallableImpl.call(CeWorkerCallableImpl.java:35)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```